feat(#976): Create contributor onboarding space (#992)
medic/cht-docs: content/en/contribute/medic/onboarding/technical-resources.md

Keep in mind that this is a living document, and every contributor is encouraged to add to it when identifying learning opportunities that can set them up for success with the CHT. The main goal here is to provide the technical people who are new to the CHT with a list of resources that could be beneficial for them in both the short and long term. However, the expectation is not to complete every course listed in this document but instead to share some resources that other contributors found helpful. It is valid if you prefer to use other training material, as long as you cover the skill/topic during the onboarding. That means you will need to learn JS, Docker, CouchDB, and others mentioned in the “Skills” column, as you will usem them in your day-to-day activities, and the list provided in the “Resources” column is meant to support you in this. It is OK if you prefer to learn with alternative resources. 

{{% alert title="Note" color="info" %}}

> Some of the suggestions below could be for internal courses and/or material created and maintained by other teammates at Medic, free resources on the internet, or paid courses, which for Medic employees could be considered as part of the [Professional Development Benefit](https://www.notion.so/Professional-Development-f5fc299ecf984dacbd65100af100ed14) at Medic.

Reviewer: This is generally our preferred terminology for such cases:

```suggestion
Some of the suggestions below could be for internal courses and/or material created and maintained by other teammates at Medic, free resources on the internet, or paid courses, which for Medic teammates could be considered as part of the [Professional Development Benefit](https://www.notion.so/Professional-Development-f5fc299ecf984dacbd65100af100ed14) at Medic.
```

...I think you got the text from elsewhere, so not really your fault. This is just a good chance to update it.

{{% /alert %}}

## Exercises

> There is a Slack channel *#exercism* where engineers and tech enthusiats meet and discuss a Javascript exercise from [exercism.org](https://exercism.org/) on a weekly basis. The primary goal of this channel is to discuss with your teammates and have a fun, open discussion on how the provided solutions can be improved, while also having the opportunity to connect with people from other teams. Even though this is optional, it is highly encouraged to join and participate frequently.

Reviewer: We're on TypeScript now! :smile: 

```suggestion
There is a Slack channel *#exercism* where engineers and tech enthusiats meet and discuss Javascript and TypeScript exercises from [exercism.org](https://exercism.org/) on a weekly basis. The primary goal of this channel is to discuss with your teammates and have a fun, open discussion on how the provided solutions can be improved, while also having the opportunity to connect with people from other teams. Even though this is optional, it is highly encouraged to join and participate frequently.
```

---
title: "Technical Resources"
linkTitle: "Technical Resources"
weight: 3
description: >
  Resources to get started as a contributor to the CHT
---

{{% alert title="Note" color="info" %}}
This page also contains Medic internal documentation links that are only accessible to Medic team members.
{{% /alert %}}

This page contains materials that will help a CHT Contributor to master all the technical details they need to be successful when building tools related to the CHT. For Medic team members, this content is complimentary on a technical level to the Onboarding documents shared by the Internal Operations team. 

> Keep in mind that this is a living document, and every contributor is encouraged to add to it when identifying learning opportunities that can set them up for success with the CHT. The main goal here is to provide the technical people who are new to the CHT with a list of resources that could be beneficial for them in both the short and long term. However, the expectation is not to complete every course listed in this document but instead to share some resources that other contributors found helpful. It is valid if you prefer to use other training material, as long as you cover the skill/topic during the onboarding. That means you will need to learn JavaScript, TypeScript, Docker, CouchDB, and others mentioned in the “Skills” column, as you will usem them in your day-to-day activities, and the list provided in the “Resources” column is meant to support you in this. It is OK if you prefer to learn with alternative resources. 

Reviewer 2: ```suggestion
Keep in mind that this is a living document, and every contributor is encouraged to add to it when identifying learning opportunities that can set them up for success with the CHT. The main goal here is to provide the technical people who are new to the CHT with a list of resources that could be beneficial for them in both the short and long term. However, the expectation is not to complete every course listed in this document but instead to share some resources that other contributors found helpful. It is valid if you prefer to use other training material, as long as you cover the skill/topic during the onboarding. That means you will need to learn JavaScript, TypeScript, Docker, CouchDB, and others mentioned in the “Skills” column, as you will use them in your day-to-day activities, and the list provided in the “Resources” column is meant to support you in this. It is OK if you prefer to learn with alternative resources. 
```

| **NodeJS and NPM** | [The Complete Node.JS developer course](https://www.udemy.com/course/the-complete-nodejs-developer-course-2/) <br> [Understanding NPM - Node.js Package Manager](https://www.udemy.com/course/understanding-npm/)|
| **Angular** | [Angular 12 - Complete Beginner's Guide 2021](https://www.udemy.com/course/learning-angular/)|
| **Offline First** | [Offline Web Applications](https://www.youtube.com/playlist?list=PLAwxTw4SYaPmTSxtOWyJVKTUaNBGze2ed) |
| **Web Architecture** | [Web Architecture 101](https://medium.com/storyblocks-engineering/web-architecture-101-a3224e126947) |
| **Web Browsers** | [Inside look at modern web browsers, part I](https://developer.chrome.com/blog/inside-browser-part1/) <br> [Inside look at modern web browsers, part II](https://developer.chrome.com/blog/inside-browser-part2/) <br> [Inside look at modern web browsers, part III](https://developer.chrome.com/blog/inside-browser-part3/) <br> [Inside look at modern web browsers, part IV](https://developer.chrome.com/blog/inside-browser-part4/)|

> | **Web Performance** | [High Performance Browser Networking](https://hpbn.co/)|

Reviewer 2: I expect this table will grow over time. Would it be more accessible to use headers and lists instead of table format?

eg:

### CHT Local Environment and Docs

- [Set up your local development environment]({{% ref "dev-environment" %}})
- [Go through CHT Documentation]({{% ref "/" %}}) <br> [Get familiar with our Development Workflow]({{% ref "/contribute/code/workflow" %}})

... etc

Author: I prefer the table style for the readability and conciseness of the page. Yes, we might add links in the future, but it will only happen sometimes. Let's keep it this way for now and change the style if we get to a point where it is not readable anymore.
